plugin/kubernetes/controller.go

```diff
@@ -130,15 +133,35 @@ func newdnsController(ctx context.Context, kubeClient kubernetes.Interface, opts
 	}
 
 	if opts.initEndpointsCache {
-		dns.epLister, dns.epController = object.NewIndexerInformer(
-			&cache.ListWatch{
+		var (
+			apiObj            runtime.Object
+			listWatch         *cache.ListWatch
+			toFuncFunc        func(bool) object.ToFunc
+			recordLatencyFunc object.RecordLatencyFunc
+		)
+		if opts.useEndpointSlices {
+			apiObj = &discovery.EndpointSlice{}
+			listWatch = &cache.ListWatch{
+				ListFunc:  endpointSliceListFunc(ctx, dns.client, api.NamespaceAll, dns.selector),
+				WatchFunc: endpointSliceWatchFunc(ctx, dns.client, api.NamespaceAll, dns.selector),
+			}
+			toFuncFunc = object.EndpointSliceToEndpoints
```

Comment:
why FuncFunc?

Reply:
It's a function that returns function that returns a function.  IOW, a function that returns an object.ToFunc, not an object.ToFunc itself. Made sense to me, but I'll rename it to just `to`.